### PR TITLE
PR: Update update_fa5 command to rename internal font names during update

### PR DESCRIPTION
**Pull request by darkvertex**
In an effort to prevent further mistakes, I took the core code from https://github.com/chrissimpkins/fontname.py and made it into a utility function that then renames "regular" and "solid" flavours of FA5 to prevent issues like #107 .

This means that the command technically introduces a requirement for the "fonttools" module. Since it is a "dev" command I did not include it in the setup.py's `install_requires` but I did a lot of reading and it seems very tricky to declare the dependency for the command only. There's an `extra_requires` thing but that only works for package "entry_points" (which we don't use.) If you know a better way, I'm all ears.

For now, it'll raise an exception if the import fails.

### CODE REVIEW

File: setupbase.py [hunk 1]
```diff
@@ -129,13 +189,24 @@ def run(self):
...
-            with open(fontPath, 'w+') as f:
+            self.__print('Dumping updated "%s" font: %s' % (style, font_path))
+            with open(font_path, 'w+') as f:
                 f.write(data)
 
+            # Fix to prevent repeated font names:
+            if style in ('regular', 'solid'):
+                newName = str("Font Awesome 5 Free %s") % style.title()
+                self.__print('Renaming font to "%s" in: %s' % (newName, font_path))
+                rename_font(font_path, newName)
```
Comment (ccordoba12) [inline setupbase.py]: Please change this line to

```python
if ttlib is not None:
    rename_font(font_path, new_name)
else:
    sys.exit(
        "This special command requires the module 'fonttools': "
        "https://github.com/fonttools/fonttools/")
```

File: setupbase.py [hunk 2]
```diff
@@ -129,13 +189,24 @@ def run(self):
...
-            hashes[style] = hashlib.md5(data).hexdigest()
-            self.__print('Dumping updated "%s" font: %s' % (style, fontPath))
-            with open(fontPath, 'w+') as f:
+            self.__print('Dumping updated "%s" font: %s' % (style, font_path))
+            with open(font_path, 'w+') as f:
                 f.write(data)
 
+            # Fix to prevent repeated font names:
+            if style in ('regular', 'solid'):
+                newName = str("Font Awesome 5 Free %s") % style.title()
```
Comment (ccordoba12) [inline setupbase.py]: Please change `newName` here to `new_name`.

Comment (darkvertex) [reply setupbase.py]: Woops! Where I work our style guide is to use camelCase. I know this project uses snake_case and I made a mental effort to make it so, but I guess habit got the best of me! hehe. I'll edit it.

File: setupbase.py [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
 import hashlib
 import zipfile
+
+try:
+    from fontTools import ttLib
+    # from fontTools.misc.py23 import tounicode, unicode
+except ImportError:
+    sys.exit(
+        "This special command requires the module 'fonttools': "
+        "https://github.com/fonttools/fonttools/")
```
[the remark this answers is not in the document]
Comment (darkvertex) [reply setupbase.py]: Ah! True. Thanks for the tip.

**Comment by darkvertex** [comment]
@ccordoba12 I addressed your feedback. Tests are clearing now. Thanks for the review.

File: setupbase.py [hunk 4]
```diff
@@ -1,10 +1,18 @@
 # -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
Comment (ccordoba12) [inline setupbase.py]: Why is this line necessary?

Comment (darkvertex) [reply setupbase.py]: I'm not 100% sure to be honest. I had kept it when copying code from: https://github.com/chrissimpkins/fontname.py/blob/master/fontname.py#L20
I did very minimal editing to make it usable as a helper function, but the bulk of the logic is as-is.

I just tried running without it and it seems fine, so I guess it's not really needed.
